File: java/Q541_reverse_string_ii.java
public class Q541_reverse_string_ii {
    public String reverseStr(String s, int k) {
        String ans = "";
        int p = 0;

        for (int i = 0; i <= s.length()-1; i += k) {
            String sub = s.substring(i,Math.min(i+k,s.length()));
            String sub_reverse = "";

            if (p % 2 == 0) {
                for (int j = sub.length()-1; j >= 0; j--) {
                    sub_reverse += sub.charAt(j);
                }
                ans += sub_reverse;
            } else {
                ans += sub;
            }

            p++;
        }

        return ans;
    }
}

File: python3/Q541_reverse_string_ii.py
class Solution:
    def reverseStr(self, s: str, k: int) -> str:
        ans = ""
        p = 0

        for i in range(0,len(s),k):
            sub = ""
            sub = s[i:i+k]
            if p % 2 == 0:
                sub = sub[::-1]
            ans += sub
            p += 1

        return ans